add-on/src/contentScripts/ipfs-proxy/page.js

```diff
@@ -46,7 +45,4 @@ function createWindowIpfs () {
   return freeze(proxyClient)
 }
 
-// TODO: we should remove Buffer and add support for Uint8Array/ArrayBuffer natively
-// See: https://github.com/ipfs/interface-ipfs-core/issues/404
-window.Buffer = window.Buffer || _Buffer
```

Comment:
I think we should add support for `TypedArray` and `ArrayBuffer` before we ship this.

I'd be interested to know if [this](https://github.com/ipfs-shipyard/js-ipfs-http-client-lite/blob/master/src/add/normalise-input.js) covers all the bases.

Reply:
Yeah, I've been thinking about it and (C) feels like a right way to do it.

Something I realized is that the first thing people do when they play with API is to add some text (`String`) to IPFS.  What if we remove friction of converting to bytes and accept `String` input and convert it internally to a typed array?

afaik something like https://encoding.spec.whatwg.org would do:

```js
const enc = new TextEncoder() // implicit utf-8
const bytes = enc.encode('This string will be converted to a Uint8Array'))
```

Reply:
Yes, I've wanted this for a while and allowed it in https://github.com/ipfs-shipyard/ipfsx/blob/master/API.md#example-1

Reply:
The tricky bit with it is destinguishing between content and a path in the file system, which I think is supported somewhere

Reply:
yeah thats the problem addFromFs

Reply:
Is `addFromFs` the only place?
I think it is quite intuitive special case where `String` is interpreted as file path. It could run the generic `normalizeInput` as every other command and then convert bytes back to file path internally.

Reply:
Previously we had everything in the add, addReadable, addPullstream when we introduced addFromFs i think we kinda separated the `/file/path` input to this method only need further investigation